packages/@apphosting/adapter-nextjs/src/utils.ts

```diff
@@ -1,21 +1,27 @@
+import { spawnSync } from "child_process";
 import fsExtra from "fs-extra";
-import { PHASE_PRODUCTION_BUILD } from "./constants.js";
-import { ROUTES_MANIFEST } from "./constants.js";
+import { createRequire } from "node:module";
+import { join, relative, normalize } from "path";
 import { fileURLToPath } from "url";
-import { OutputBundleOptions } from "./interfaces.js";
 import { stringify as yamlStringify } from "yaml";
-import { spawnSync } from "child_process";
 
-import { join, relative, normalize } from "path";
+import { PHASE_PRODUCTION_BUILD } from "./constants.js";
+import { ROUTES_MANIFEST } from "./constants.js";
+import { OutputBundleOptions, RoutesManifest } from "./interfaces.js";
+import { NextConfigComplete } from "next/dist/server/config-shared.js";
 
-import type { RoutesManifest } from "./interfaces.js";
 // fs-extra is CJS, readJson can't be imported using shorthand
 export const { move, exists, writeFile, readJson } = fsExtra;
 
-export async function loadConfig(cwd: string) {
+export const DEFAULT_COMMAND = "npm";
+
+export async function loadConfig(cwd: string): Promise<NextConfigComplete> {
   // dynamically load NextJS so this can be used in an NPX context
+  const require = createRequire(import.meta.url);
```

Comment:
Why do we need this change? We might have discussed this but I forgot the reason

Reply:
Yeah we discussed this a few weeks ago and my memory was also a bit fuzzy, but we need this in order to locate the package in the root node_modules folder (which is relevant for monorepo contexts).

Reply:
Is there any additional documentation we can link to in this line comment?

Reply:
Sorry not to be too pedantic but this comment explains _what_ this does as opposed to _why_ we need it.

Can we have a comment here explaining the motivation behind this statement? eg "Intentionally use ESM module resolution here to support blah blah blah"

See [go/tott/563](https://goto.google.com/tott/563)

Reply:
A good litmus test is: Would someone who's not you, a year from now, understand why we need this statement based on the line comment? Would they be able to gather sufficient context either from the code, the comment, or linked docs to own it going forward?